src/main/java/de/tum/cit/aet/artemis/core/web/CourseResource.java

```diff
@@ -1453,4 +1455,23 @@ public ResponseEntity<Long> getNumberOfAllowedComplaintsInCourse(@PathVariable L
         long unacceptedComplaints = complaintService.countUnacceptedComplaintsByParticipantAndCourseId(participant, courseId);
         return ResponseEntity.ok(Math.max(complaintService.getMaxComplaintsPerParticipant(course, participant) - unacceptedComplaints, 0));
     }
+
+    @GetMapping("courses/{courseId}/existing-exercise-details")
+    @EnforceAtLeastEditorInCourse
+    public ResponseEntity<CourseExistingExerciseDetails> getExistingExerciseDetails(@PathVariable Long courseId, @RequestParam(defaultValue = "false") Boolean includeShortNames) {
```

Comment:
I don't really see the benefit of the boolean parameter. I think always sending the shortNames should not be too bad...
If you want to keep the boolean:
```suggestion
    public ResponseEntity<CourseExistingExerciseDetails> getExistingExerciseDetails(@PathVariable Long courseId, @RequestParam(defaultValue = "false") boolean includeShortNames) {
```

Reply:
Even though the amount of data that is transferred should be manageable, I do still think that it is best practice to keep the transferred data as small as possible, if we would ignore it at all endpoints we might run into problems aswell.
The shortnames will only be needed for the creation of programming exercises.

>     /**
>      * GET courses/{courseId}/existing-exercise-details: Get the exercise names and shortNames of all exercises in the given course.
>      *
>      * @param courseId          of the course for which all exercise names (and optionally shortNames) should be fetched
>      * @param includeShortNames should be true for programming exercises and false for other exercise types as the shortName is not used there
>      * @return {@link CourseExistingExerciseDetails} with the exerciseNames and optionally shortNames of the course
>      */